static/panes/device-view.ts

```diff
@@ -0,0 +1,378 @@
+// Copyright (c) 2022, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import {FontScale} from '../widgets/fontscale';
+import * as monaco from 'monaco-editor';
+import _ from 'underscore';
+import $ from 'jquery';
+import * as colour from '../colour';
+import {ga} from '../analytics';
+import * as monacoConfig from '../monaco-config';
+import {PaneRenaming} from '../widgets/pane-renaming';
+import TomSelect from 'tom-select';
+import GoldenLayout from 'golden-layout';
+import {Hub} from '../hub';
+import {MonacoPane} from './pane';
+import {DeviceAsmCode, DeviceAsmState} from './device-view.interfaces';
+import {MonacoPaneState} from './pane.interfaces';
+
+type DecorationEntry = {
+    linkedCode: any[];
```

Comment:
Is this the first pane that uses decorations to be ported to TS? I feel like it ought to have a better type. It will probably be a built-in Monaco editor one